app/controllers/ReservationController.java

```diff
@@ -2,84 +2,63 @@
-//
-//    Date date = boundForm.get().getDate();
-//
-//    String time = boundForm.get().getTime();
-//
-//    int people = boundForm.get().getPeople();
-//
-//    int numOfFreeTables = RestaurantTable.getTables(restaurantId, people).size();
-//
-//
-//        if(restaurantName == null){
-//                Logger.info("987987987987987987987");
-//            int restaurantId1 = 5;
-//            String restaurantName1 = "L A P A";
-//            String restaurantImage1 = "http://ceka.ba/images/ceka1-b.jpg";
-//            Date date1 = new Date();
-//            String time1 = "02:14 PM";
-//            int people1 = 7;
-//            int numOfFreeTables1 = 22;
-//            List<String> bestTimes1 = new ArrayList<>();
-//            bestTimes1.add(time.toString());
-//            bestTimes1.add("10:30 AM");
-//            bestTimes1.add("11:00 AM");
-//            bestTimes1.add("11:30 AM");
-//            ReservationResponse reservationResponse1 = new ReservationResponse(numOfFreeTables1,bestTimes1,restaurantId1,restaurantName1, restaurantImage1,date1,people1,time1);
-//
-//            return ok(Json.toJson(reservationResponse1));
-//                }
-//
-//
-//
-//
-//        List<String> bestTimes = new ArrayList<>();
-//        bestTimes.add(time.toString());
-//        bestTimes.add("10:30 AM");
-//        bestTimes.add("11:00 AM");
-//        bestTimes.add("11:30 AM");
-//
-//        SimpleDateFormat sdf = new SimpleDateFormat("dd.MM.yyyy ");
-//        String d = sdf.format(date);
-//
-//        ReservationResponse reservationResponse = new ReservationResponse(numOfFreeTables,bestTimes,restaurantId,restaurantName, restaurantImage,date,people,time);
-//
-//
-//        return ok(Json.toJson(reservationResponse));
-//
-        return ok();
+
+        Form<Reservation> boundForm = Reservation.reservationForm.bindFromRequest();
+
+    Long restaurantId = Long.valueOf(boundForm.bindFromRequest().field("restaurantId").value());
+    Restaurant restaurant = reservationService.getRestaurantById(restaurantId);
+    String restaurantName = restaurant.getName();
+    String restaurantImage = restaurant.getImage();
+    int people = boundForm.get().getNumberOfPeople();
+    int numOfFreeTables = reservationService.getTablesForSelectedNumberOfPeople(restaurantId, people).size();
+    String date = boundForm.get().getDate();
+    String time = boundForm.get().getTime();
+        Timestamp timestamp = Timestamp.valueOf(date + " " + time + ":00.0");
+        Long timestamp2 = timestamp.getTime();
+        if(timestamp2 < System.currentTimeMillis()){
+            return badRequest(Response.errorResponse("Can't make reservation for past time."));
+        }
+
+        List<String> bestTimes = reservationService.getBestTimes(date, timestamp2, restaurantId, people);
+
+        ReservationResponse reservationResponse = new ReservationResponse(numOfFreeTables,bestTimes,restaurantId,restaurantName, restaurantImage,timestamp,people,time);
```

Comment:
Service that checks for available tables should return this `ReservationResponse`, service should accept request body as argument (in this case `Reservation`), and build this response.